tests/server/test_loggers.py

```diff
@@ -0,0 +1,178 @@
+from concurrent.futures import ThreadPoolExecutor
+
+import pytest
+from deepsparse import PythonLogger
+from deepsparse.engine import Context
+from deepsparse.server.build_logger import build_logger
+from deepsparse.server.config import EndpointConfig, MetricFunctionConfig, ServerConfig
+from deepsparse.server.server import _add_endpoint, _build_app
+from fastapi.testclient import TestClient
+from tests.helpers import find_free_port
+from tests.utils import mock_engine
+
+
+LIST_LOGGER_IDENTIFIER = "tests/deepsparse/loggers/helpers.py:ListLogger"
+
+EXPECTED_LOGS_SYSTEM = {"category:MetricCategories.SYSTEM": 8}
+EXPECTED_LOGS_REGEX = {"pipeline_inputs.identity": 2, "pipeline_outputs.identity": 2}
+EXPECTED_LOGS_MULTIPLE_TARGETS = {
+    "pipeline_inputs.sequences.identity": 2,
+    "engine_inputs.identity": 2,
+}
+EXPECTED_LOGS_TARGET_LOGGER = [
+    {**EXPECTED_LOGS_SYSTEM, **EXPECTED_LOGS_MULTIPLE_TARGETS},
+    {
+        **EXPECTED_LOGS_SYSTEM,
+        **{"pipeline_inputs.sequences.identity": 0, "engine_inputs.identity": 2},
+    },
+]
+
+DATA_LOGGER_CONFIG_REGEX = {"re:.*pipeline*.": [MetricFunctionConfig(func="identity")]}
+DATA_LOGGER_CONFIG_MULTIPLE_TARGETS = {
+    "pipeline_inputs.sequences": [MetricFunctionConfig(func="identity")],
+    "engine_inputs": [MetricFunctionConfig(func="identity")],
+}
+DATA_LOGGER_CONFIG_TARGET_LOGGER = {
+    "pipeline_inputs.sequences": [
+        MetricFunctionConfig(func="identity", target_loggers=["logger_1"])
+    ],
+    "engine_inputs": [MetricFunctionConfig(func="identity")],
+}
+
+
+def _test_logger_contents(leaf_logger, expected_logs):
+    for expected_log_content in list(expected_logs.keys()):
+        i = 0
+        for log in leaf_logger.calls:
+            if expected_log_content in log:
+                i += 1
+        assert expected_logs[expected_log_content] == i
+
+
+@pytest.mark.parametrize(
+    "loggers, data_logger_config, expected_logs_content",
+    [
+        ({}, [], None),  # Test default logger (no `loggers` specified)
+        (
+            {"logger_1": {"path": LIST_LOGGER_IDENTIFIER}},
+            [],
+            EXPECTED_LOGS_SYSTEM,
+        ),  # Make sure that we log system logs only
+        (
+            {"logger_1": {"path": LIST_LOGGER_IDENTIFIER}},
+            DATA_LOGGER_CONFIG_REGEX,
+            {**EXPECTED_LOGS_SYSTEM, **EXPECTED_LOGS_REGEX},
+        ),  # Make sure we can use regex to target specific identifiers
+        (
+            {"logger_1": {"path": LIST_LOGGER_IDENTIFIER}},
+            DATA_LOGGER_CONFIG_MULTIPLE_TARGETS,
+            EXPECTED_LOGS_MULTIPLE_TARGETS,
+        ),  # Test multiple targets
+        (
+            {
+                "logger_1": {"path": LIST_LOGGER_IDENTIFIER},
+                "logger_2": {"path": LIST_LOGGER_IDENTIFIER},
+            },
+            DATA_LOGGER_CONFIG_TARGET_LOGGER,
+            EXPECTED_LOGS_TARGET_LOGGER,
+        ),  # One function metric with target loggers
+    ],
+)
+class TestLoggers:
```

Comment:
Those tests are quite clunky and heavy-weight. @corey-nm could we sync offline and maybe discuss how I could improve them?